migrations/dimensions-config/bf6308af80aa.py

```diff
@@ -0,0 +1,402 @@
+
+    # Fill seq_num column with the lowest value of matching exposure.seq_num.
+    #
+    # Query for correlated update:
+    #
+    # UPDATE
+    #     visit
+    # SET
+    #     seq_num = (
+    #         SELECT
+    #             MIN(exposure.seq_num)
+    #         FROM
+    #             exposure
+    #             JOIN visit_definition ON visit_definition.exposure = exposure.id
+    #             and visit_definition.instrument = exposure.instrument
+    #         WHERE
+    #             visit_definition.instrument = visit.instrument,
+    #             visit_definition.visit = visit.id
+    #     )
+
+    visit = sa.schema.Table("visit", metadata, autoload_with=bind, schema=schema)
+    exposure = sa.schema.Table("exposure", metadata, autoload_with=bind, schema=schema)
+    visit_def = sa.schema.Table("visit_definition", metadata, autoload_with=bind, schema=schema)
+
+    min_seq = (
+        sa.select(sa.sql.functions.min(exposure.columns["seq_num"]))
+        .select_from(exposure.join(visit_def))
+        .where(
+            sa.sql.expression.and_(
+                visit_def.columns["instrument"] == visit.columns["instrument"],
+                visit_def.columns["visit"] == visit.columns["id"],
+            )
+        )
+    )
+    sql = visit.update().values(seq_num=min_seq.scalar_subquery())
+    op.execute(sql)
+
+
+def _migrate_exposure(has_simulated: bool) -> None:
+    """Update exposure table.
+
+    Parameters
+    ----------
+    has_simulated : `bool`
+        Value for the new ``has_simulated`` column.
+    """
+    mig_context = context.get_context()
+    bind = mig_context.bind
+    schema = mig_context.version_table_schema
+    metadata = sa.schema.MetaData(bind, schema=schema)
+
+    _LOG.info("migrating exposure table")
+    with op.batch_alter_table("exposure", schema=schema) as batch_op:
+        batch_op.add_column(sa.Column("seq_start", sa.BigInteger))
+        batch_op.add_column(sa.Column("seq_end", sa.BigInteger))
+        batch_op.add_column(sa.Column("azimuth", sa.Float))
+        batch_op.add_column(sa.Column("has_simulated", sa.Boolean))
+
+    table = sa.schema.Table("exposure", metadata, autoload_with=bind, schema=schema)
+    op.execute(
+        table.update().values(
+            seq_start=table.columns["seq_num"], seq_end=table.columns["seq_num"], has_simulated=has_simulated
+        )
+    )
+
+
+def _migrate_dimensions_json() -> None:
+    """Updates dimensions definitions in dimensions.json."""
+
+    mig_context = context.get_context()
+    schema = mig_context.version_table_schema
+    attributes = ButlerAttributes(mig_context.bind, schema)
+
+    def update_config(config: Dict) -> Dict:
+
+        version = config.get("version")
+        if version != 1:
+            raise ValueError(f"unexpected version number: {version}")
+
+        config["version"] = 2
+
+        instrument = config["elements"]["instrument"]
+        instrument["metadata"].insert(
```

Comment:
I thought about making a copy of a complete `dimensions.yaml` but I like diff-style updates better in this casae. It is easier to relate these diffs to the changes made to the actual schema. I think it  should also be easier to port this to other branches/namespaces. For more complicated updates it may indeed be simpler to have a complete YAML loaded at once.